src/core/reactor.cc

```diff
@@ -4332,7 +4332,7 @@ void smp::configure(const smp_options& smp_opts, const reactor_options& reactor_
         hugepages_path = smp_opts.hugepages.get_value();
     }
     auto mlock = false;
-    if (smp_opts.lock_memory) {
+    if (smp_opts.lock_memory && !reactor_opts.overprovisioned) {
         mlock = smp_opts.lock_memory.get_value();
```

Comment:
It's default off, so why this extra work?

If someone specified both --overprovisioned (which really should have been named --overcommitted) and --lock-memory 1, we can assume they know what they're doing. Maybe only the CPU is overcommitted.

Reply:
ok, I agree. Then I think we can tell the same for `thread_afinity` option. A user who knows what he is doing can overcommit memory and pin vCPUs. So I think it can make sense to only log a warning in both cases. wdyt?